lib/autsr.gi

```diff
@@ -170,10 +170,74 @@ end);
 
 # step through intermediate subgroups to find subgroup that satisfies
 # condition
-BindGlobal("SubgroupConditionAbove",function(G,cond,S)
-local c,hom,q,a,t,int,bad,have,ups,up,new,u;
+InstallGlobalFunction(SubgroupConditionAbove,function(G,cond,S)
+local c,hom,q,a,b,i,t,int,bad,have,ups,up,new,u,good,abort;
+
+  Info(InfoMorph,2,"SubgroupAbove ",IndexNC(G,S));
+
+  # first try, how far `SubgroupProperty` goes
+  good:=[];
+  bad:=[];
+  b:=0;
+  abort:=false;
+  c:=SubgroupProperty(G,
+    function(elm)
+      if abort then
+        return true; # are we bailing out since it behaves too badly?
+      elif cond(elm) then
+        Add(good,elm);
+        S:=ClosureGroup(S,elm); # remember
+        Info(InfoMorph,3,"New element ",IndexNC(G,S));
+        b:=0;
+        return true;
+      else
+        Add(bad,elm);
+        b:=b+1;
+        if b*2000>IndexNC(G,S) then
+          abort:=true;
+        fi;
+        return false;
+      fi;
+    end,S);
+
+  if abort=false then 
+    # we actually found the subgroup
+    Info(InfoMorph,2,"SubgroupProperty finds ",IndexNC(G,c));
+    return c;
+  fi; 
+
+  Info(InfoMorph,2,"intermediate improvement ",IndexNC(G,S));
+
+  good:=false;
+  if IndexNC(G,S)<=10^5 then
```

Comment:
@hulpke _If_ you think it would be possible to give some brief explanation for why you have chosen the 'magic constants' that you have in this PR, in a way that could make future maintenance easier, then I would appreciate that. There is this line, the `b*2000` above, and the updated `lim` below, that seem to be things of this kind.

I envisage something like: `with 10^5, this typically takes no more than a few seconds`. That way, if in the future `10^6` would take just a few seconds, a future maintainer could choose to increase the limit.

If you don't think that's a good idea/not useful/don't want to, then that's okay.

Reply:
I've added some text explaining the numbers. The issue of the 10^5 is more memory than time. The 2000 is 1/20 percent as both establishing a "really small success rate" versus not trying in vain for too long.